lang/gjs/package.json

```
@@ -29,7 +29,9 @@
         "./gobject": "./src/gobject.ts",
         "./process": "./src/process.ts",
         "./time": "./src/time.ts",
-        "./variable": "./src/variable.ts"
+        "./variable": "./src/variable.ts",
+        "./_app": "./src/_app.ts",
+        "./_astal": "./src/_astal.ts"
```

These are not public modules, which is why they start with an underscore.
Why do you need them exported?

---

I made my own `astalify` function as a subclass for gtk4 here: https://github.com/matt1432/nixos-configs/blob/cdcdd23e7ff18b136bc246fe0ad80830cb3003a8/modules/ags/gtk4/widget/subclasses/astalify.ts#L13

---

I know the gtk3 astalify subclasses the widgets as well, but I always wondered how this works. Many Widgets are declared as final, so subclassing them should not work, or at least registering them as a GType should throw an error.

---

I tried all of the subclasses I did and they all worked. All the base Widgets aren't final, it's just the very specific ones that are afaik

---

Many widget in gtk3 and almost all widgets in gtk4 are final, they just aren't marked as such in gjs, so you can subclass them even though they are final. But that's beyond the point.

---

Huh that's weird. So would this be considered a GJS bug then?